controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -76,6 +74,41 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// - ocmApiUrl: Optional, defaults to 'https://api.openshift.com'
 	// +optional
 	CredentialsSecretRef *corev1.LocalObjectReference `json:"credentialsSecretRef,omitempty"`
+
+	// Network config for the ROSA-HCP cluster.
+	// +optional
+	Network *NetworkSpec `json:"network,omitempty"`
+
+	// The compute machine type to use, for example `r5.xlarge`. Machine type ref; https://aws.amazon.com/ec2/instance-types/
+	// +optional
+	ComputeMachineType string `json:"computeMachineType,omitempty"`
```

Comment:
```suggestion
	// The compute machine type to use, for example `r5.xlarge`. Machine type ref; https://aws.amazon.com/ec2/instance-types/
	// +optional
	MachineType string `json:"computeMachineType,omitempty"`
```

Reply:
We should probably use whatever AWSMachine has, which iirc is `InstanceType`?

Reply:
I would keep it machineType to match OCM API

Reply:
`Machine` is an overloaded term within Cluster API, let's keep it consistent with `AWSMachine.Spec`, which uses `InstanceType`